File: Greedy_Algorithm/Python/백준_그리디알고리즘_거스름돈(5585).py
# -*- coding: utf-8 -*-
import sys
N = int(sys.stdin.readline())
money = 1000-N
arr = [500,100,50,10,5,1]
count = 0
for i in range(6):
    if money>=arr[i]:
        count += money//arr[i]
        money %= arr[i]
    else:
        continue
print(count)

## / 는 float 형태 , // 는 int 형태

"""
for arr in arrs:
    if money == 0:
        break
    if money // arr > 0:
        count += money // arr
        money %= arr
"""

File: Greedy_Algorithm/C++/백준_그리디알고리즘_거스름돈(5585).cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    int money;
    int count = 0;
    cin >> money;
    money = 1000 - money;
    int arr[6] = { 500,100,50,10,5,1 };
    for (int i = 0; i < 6; i++) {
        if (money >= arr[i]) {
            count += money / arr[i];
            money %= arr[i];
        }
        else continue;
    }
    cout << count << endl;
    return 0;
}
